### ISSUE TITLE
ScrollView doesnt work with TimePicker inside it Xamarin Forms

### ISSUE THREAD

**Issue by vsfeedback** [xamarin/Xamarin.Forms#9103]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/875358/scrollview-doesnt-work-with-timepicker-inside-it-x.html)._

---
I have found bug When you have timePicker nested inside ScrollView it doesn't work. How can I replace it, or what is your suggestion to solve it. Error
**Android.Views.InflateException: 'Binary XML file line #20: Binary XML file line #20: Error inflating class android.widget.TimePicker'**

---
#### Visual Studio Feedback System on 1/6/2020, 00:18 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Martin Evans on 1/6/2020, 09:32 AM: 

<p>I have just updated to 16.4 and am now also getting the same problem. For me this happens on a Galaxy S5 phone, but not when using an Android Emulator.</p>

---
### Original Solutions
(no solutions)

**Comment by rozek1szymon** [comment]
I use Samsung A5, for this moment I cant use another phone :/. Can somebody check it on other phones

**Comment by samhouts** [comment]
Can you please attach a small project that demonstrates this issue? Thanks!

**Comment by rozek1szymon** [comment]
> Can you please attach a small project that demonstrates this issue? Thanks!
`<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:d="http://xamarin.com/schemas/2014/forms/design"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
             mc:Ignorable="d"
             x:Class="MeetAppMobile.Views.CalendarPage">
    <StackLayout>
        <ScrollView>
            <TimePicker>
                
            </TimePicker>
        </ScrollView>
    </StackLayout>
</ContentPage>`

**Comment by rozek1szymon** [comment]
`<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:d="http://xamarin.com/schemas/2014/forms/design"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
             mc:Ignorable="d"
             x:Class="MeetAppMobile.Views.CalendarPage">
    <StackLayout>
        <ScrollView>
            <TimePicker>             
            </TimePicker>
        </ScrollView>
    </StackLayout>
</ContentPage>`

**Comment by rozek1szymon** [comment]
Ok, I solved my problem it needed to be downgradeed the Xamarin.Forms from 4.4.0.991265 to 4.3.0.991250. It seems that, on some models like Samsung SM-A500FU, or like someone said above Galaxy S5, TimePicker inside ScrollView throws error. I hope you will solve this issue, thanks.

**Comment by martinjevans** [comment]
I can confirm that 4.3.0.991250 does solve the problem on the Samsung Galaxy S5.

**Comment by lassana** [comment]
We also got this (full stacktrace below). Any workaround would be highly appreciated.

<details> 
  <summary>stacktrace</summary>

```
Java.Lang.RuntimeException

Message: Binary XML file line #20: Binary XML file line #20: Error inflating class android.widget.TimePicker
Java.Interop.JniEnvironment+InstanceMethods.CallNonvirtualVoidMethod(Java.Interop.JniObjectReference instance, Java.Interop.JniObjectReference type, Java.Interop.JniMethodInfo method, Java.Interop.JniArgumentValue* args) in <72c0737127e04a94b78d9dfbb1be54d7>
Java.Interop.JniPeerMembers+JniInstanceMethods.FinishCreateInstance(System.String constructorSignature, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters) in <72c0737127e04a94b78d9dfbb1be54d7>
Android.App.TimePickerDialog..ctor(Android.Content.Context context, Android.App.TimePickerDialog+IOnTimeSetListener listener, System.Int32 hourOfDay, System.Int32 minute, System.Boolean is24HourView) in <9a99ea69b8b24db49d54e5e751988a03>
Xamarin.Forms.Platform.Android.TimePickerRendererBase`1[TControl].CreateTimePickerDialog(System.Int32 hours, System.Int32 minutes) in <b229ade1a7f34fa79b33ebf5a9d82333>
Xamarin.Forms.Platform.Android.TimePickerRendererBase`1[TControl].Xamarin.Forms.Platform.Android.IPickerRenderer.OnClick() in <b229ade1a7f34fa79b33ebf5a9d82333>
Xamarin.Forms.Platform.Android.PickerManager+PickerListener.OnClick(Android.Views.View v) in <b229ade1a7f34fa79b33ebf5a9d82333>
Android.Views.View+IOnClickListenerInvoker.n_OnClick_Landroid_view_View_(System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_v) in <9a99ea69b8b24db49d54e5e751988a03>
at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.83(intptr,intptr,intptr)
--- End of managed Java.Lang.RuntimeException stack trace ---
android.view.InflateException: Binary XML file line #20: Binary XML file line #20: Error inflating class android.widget.TimePicker
android.view.LayoutInflater.inflate in LayoutInflater.java:551
android.view.LayoutInflater.inflate in LayoutInflater.java:429
android.view.LayoutInflater.inflate in LayoutInflater.java:380
android.app.TimePickerDialog.init in TimePickerDialog.java:147
android.app.TimePickerDialog.init in TimePickerDialog.java:107
crc643f46942d9dd1fff9.PickerManager_PickerListener.n_onClick(Native Method)
crc643f46942d9dd1fff9.PickerManager_PickerListener.onClick in PickerManager_PickerListener.java:30
android.view.View.performClick in View.java:5702
android.widget.TextView.performClick in TextView.java:10896
android.view.View$PerformClick.run in View.java:22533
android.os.Handler.handleCallback in Handler.java:739
android.os.Handler.dispatchMessage in Handler.java:95
android.os.Looper.loop in Looper.java:158
android.app.ActivityThread.main in ActivityThread.java:7224
java.lang.reflect.Method.invoke(Native Method)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run in ZygoteInit.java:1230
com.android.internal.os.ZygoteInit.main in ZygoteInit.java:1120
Caused by: android.view.InflateException: Binary XML file line #20: Error inflating class android.widget.TimePicker
android.view.LayoutInflater.createView in LayoutInflater.java:657
com.android.internal.policy.PhoneLayoutInflater.onCreateView in PhoneLayoutInflater.java:58
android.view.LayoutInflater.onCreateView in LayoutInflater.java:706
android.view.LayoutInflater.createViewFromTag in LayoutInflater.java:774
android.view.LayoutInflater.createViewFromTag in LayoutInflater.java:716
android.view.LayoutInflater.inflate in LayoutInflater.java:498
... 16 more
Caused by: java.lang.reflect.InvocationTargetException
java.lang.reflect.Constructor.newInstance(Native Method)
android.view.LayoutInflater.createView in LayoutInflater.java:631
... 21 more
Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'void android.widget.TimePicker$TimePickerDelegate.onRtlPropertiesChanged(int)' on a null object reference
android.widget.TimePicker.onRtlPropertiesChanged in TimePicker.java:248
android.view.View.resolvePadding in View.java:15613
android.view.ViewGroup.resolvePadding in ViewGroup.java:6877
android.view.View.initializeScrollbarsInternal in View.java:5346
android.view.View.init in View.java:4681
android.view.ViewGroup.init in ViewGroup.java:589
android.widget.FrameLayout.init in FrameLayout.java:97
android.widget.TimePicker.init in TimePicker.java:78
android.widget.TimePicker.init in TimePicker.java:74
android.widget.TimePicker.init in TimePicker.java:70
... 23 more
Message: Binary XML file line #20: Error inflating class android.widget.TimePicker
--- End of managed Java.Lang.RuntimeException stack trace ---
android.view.InflateException: Binary XML file line #20: Error inflating class android.widget.TimePicker
android.view.LayoutInflater.createView in LayoutInflater.java:657
com.android.internal.policy.PhoneLayoutInflater.onCreateView in PhoneLayoutInflater.java:58
android.view.LayoutInflater.onCreateView in LayoutInflater.java:706
android.view.LayoutInflater.createViewFromTag in LayoutInflater.java:774
android.view.LayoutInflater.createViewFromTag in LayoutInflater.java:716
android.view.LayoutInflater.inflate in LayoutInflater.java:498
android.view.LayoutInflater.inflate in LayoutInflater.java:429
android.view.LayoutInflater.inflate in LayoutInflater.java:380
android.app.TimePickerDialog.init in TimePickerDialog.java:147
android.app.TimePickerDialog.init in TimePickerDialog.java:107
crc643f46942d9dd1fff9.PickerManager_PickerListener.n_onClick(Native Method)
crc643f46942d9dd1fff9.PickerManager_PickerListener.onClick in PickerManager_PickerListener.java:30
android.view.View.performClick in View.java:5702
android.widget.TextView.performClick in TextView.java:10896
android.view.View$PerformClick.run in View.java:22533
android.os.Handler.handleCallback in Handler.java:739
android.os.Handler.dispatchMessage in Handler.java:95
android.os.Looper.loop in Looper.java:158
android.app.ActivityThread.main in ActivityThread.java:7224
java.lang.reflect.Method.invoke(Native Method)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run in ZygoteInit.java:1230
com.android.internal.os.ZygoteInit.main in ZygoteInit.java:1120
Caused by: java.lang.reflect.InvocationTargetException
java.lang.reflect.Constructor.newInstance(Native Method)
android.view.LayoutInflater.createView in LayoutInflater.java:631
... 21 more
Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'void android.widget.TimePicker$TimePickerDelegate.onRtlPropertiesChanged(int)' on a null object reference
android.widget.TimePicker.onRtlPropertiesChanged in TimePicker.java:248
android.view.View.resolvePadding in View.java:15613
android.view.ViewGroup.resolvePadding in ViewGroup.java:6877
android.view.View.initializeScrollbarsInternal in View.java:5346
android.view.View.init in View.java:4681
android.view.ViewGroup.init in ViewGroup.java:589
android.widget.FrameLayout.init in FrameLayout.java:97
android.widget.TimePicker.init in TimePicker.java:78
android.widget.TimePicker.init in TimePicker.java:74
android.widget.TimePicker.init in TimePicker.java:70
... 23 more
Message: Exception of type 'Java.Lang.ReflectiveOperationException' was thrown.
--- End of managed Java.Lang.ReflectiveOperationException stack trace ---
java.lang.reflect.InvocationTargetException
java.lang.reflect.Constructor.newInstance(Native Method)
android.view.LayoutInflater.createView in LayoutInflater.java:631
com.android.internal.policy.PhoneLayoutInflater.onCreateView in PhoneLayoutInflater.java:58
android.view.LayoutInflater.onCreateView in LayoutInflater.java:706
android.view.LayoutInflater.createViewFromTag in LayoutInflater.java:774
android.view.LayoutInflater.createViewFromTag in LayoutInflater.java:716
android.view.LayoutInflater.inflate in LayoutInflater.java:498
android.view.LayoutInflater.inflate in LayoutInflater.java:429
android.view.LayoutInflater.inflate in LayoutInflater.java:380
android.app.TimePickerDialog.init in TimePickerDialog.java:147
android.app.TimePickerDialog.init in TimePickerDialog.java:107
crc643f46942d9dd1fff9.PickerManager_PickerListener.n_onClick(Native Method)
crc643f46942d9dd1fff9.PickerManager_PickerListener.onClick in PickerManager_PickerListener.java:30
android.view.View.performClick in View.java:5702
android.widget.TextView.performClick in TextView.java:10896
android.view.View$PerformClick.run in View.java:22533
android.os.Handler.handleCallback in Handler.java:739
android.os.Handler.dispatchMessage in Handler.java:95
android.os.Looper.loop in Looper.java:158
android.app.ActivityThread.main in ActivityThread.java:7224
java.lang.reflect.Method.invoke(Native Method)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run in ZygoteInit.java:1230
com.android.internal.os.ZygoteInit.main in ZygoteInit.java:1120
Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'void android.widget.TimePicker$TimePickerDelegate.onRtlPropertiesChanged(int)' on a null object reference
android.widget.TimePicker.onRtlPropertiesChanged in TimePicker.java:248
android.view.View.resolvePadding in View.java:15613
android.view.ViewGroup.resolvePadding in ViewGroup.java:6877
android.view.View.initializeScrollbarsInternal in View.java:5346
android.view.View.init in View.java:4681
android.view.ViewGroup.init in ViewGroup.java:589
android.widget.FrameLayout.init in FrameLayout.java:97
android.widget.TimePicker.init in TimePicker.java:78
android.widget.TimePicker.init in TimePicker.java:74
android.widget.TimePicker.init in TimePicker.java:70
... 23 more
Message: Attempt to invoke interface method 'void android.widget.TimePicker$TimePickerDelegate.onRtlPropertiesChanged(int)' on a null object reference
--- End of managed Java.Lang.NullPointerException stack trace ---
java.lang.NullPointerException: Attempt to invoke interface method 'void android.widget.TimePicker$TimePickerDelegate.onRtlPropertiesChanged(int)' on a null object reference
android.widget.TimePicker.onRtlPropertiesChanged in TimePicker.java:248
android.view.View.resolvePadding in View.java:15613
android.view.ViewGroup.resolvePadding in ViewGroup.java:6877
android.view.View.initializeScrollbarsInternal in View.java:5346
android.view.View.init in View.java:4681
android.view.ViewGroup.init in ViewGroup.java:589
android.widget.FrameLayout.init in FrameLayout.java:97
android.widget.TimePicker.init in TimePicker.java:78
android.widget.TimePicker.init in TimePicker.java:74
android.widget.TimePicker.init in TimePicker.java:70
java.lang.reflect.Constructor.newInstance(Native Method)
android.view.LayoutInflater.createView in LayoutInflater.java:631
com.android.internal.policy.PhoneLayoutInflater.onCreateView in PhoneLayoutInflater.java:58
android.view.LayoutInflater.onCreateView in LayoutInflater.java:706
android.view.LayoutInflater.createViewFromTag in LayoutInflater.java:774
android.view.LayoutInflater.createViewFromTag in LayoutInflater.java:716
android.view.LayoutInflater.inflate in LayoutInflater.java:498
android.view.LayoutInflater.inflate in LayoutInflater.java:429
android.view.LayoutInflater.inflate in LayoutInflater.java:380
android.app.TimePickerDialog.init in TimePickerDialog.java:147
android.app.TimePickerDialog.init in TimePickerDialog.java:107
crc643f46942d9dd1fff9.PickerManager_PickerListener.n_onClick(Native Method)
crc643f46942d9dd1fff9.PickerManager_PickerListener.onClick in PickerManager_PickerListener.java:30
android.view.View.performClick in View.java:5702
android.widget.TextView.performClick in TextView.java:10896
android.view.View$PerformClick.run in View.java:22533
android.os.Handler.handleCallback in Handler.java:739
android.os.Handler.dispatchMessage in Handler.java:95
android.os.Looper.loop in Looper.java:158
android.app.ActivityThread.main in ActivityThread.java:7224
java.lang.reflect.Method.invoke(Native Method)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run in ZygoteInit.java:1230
com.android.internal.os.ZygoteInit.main in ZygoteInit.java:1120
```
</details>

**Comment by lassana** [comment]
This happens [on Samsung with Android 6.0.1](https://stackoverflow.com/questions/42579936/timepicker-seems-to-crash-my-app-on-samsung-devices-with-android-version-6-0-1). I ended up with removing X.F pickers and replacing them with Acr.UserDialogs.

**Comment by wouterattoptal** [comment]
I see it on Galaxy Tab A6 (aab)

```
android.view.InflateException: 
  at android.view.LayoutInflater.createView (LayoutInflater.java:640)
  at com.android.internal.policy.impl.PhoneLayoutInflater.onCreateView (PhoneLayoutInflater.java:55)
  at android.view.LayoutInflater.onCreateView (LayoutInflater.java:689)
  at android.view.LayoutInflater.createViewFromTag (LayoutInflater.java:748)
  at android.view.LayoutInflater.inflate (LayoutInflater.java:483)
  at android.view.LayoutInflater.inflate (LayoutInflater.java:415)
  at android.view.LayoutInflater.inflate (LayoutInflater.java:366)
  at android.app.TimePickerDialog.<init> (TimePickerDialog.java:135)
  at android.app.TimePickerDialog.<init> (TimePickerDialog.java:103)
  at crc643f46942d9dd1fff9.PickerManager_PickerListener.n_onClick (Native Method)
  at crc643f46942d9dd1fff9.PickerManager_PickerListener.onClick
  at android.view.View.performClick (View.java:5246)
  at android.widget.TextView.performClick (TextView.java:10626)
  at android.view.View$PerformClick.run (View.java:21256)
  at android.os.Handler.handleCallback (Handler.java:739)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:145)
  at android.app.ActivityThread.main (ActivityThread.java:6939)
  at java.lang.reflect.Method.invoke (Native Method)
  at java.lang.reflect.Method.invoke (Method.java:372)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1404)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1199)
Caused by: java.lang.reflect.InvocationTargetException: 
  at java.lang.reflect.Constructor.newInstance (Native Method)
  at java.lang.reflect.Constructor.newInstance (Constructor.java:288)
  at android.view.LayoutInflater.createView (LayoutInflater.java:614)
Caused by: java.lang.NullPointerException: 
  at android.widget.TimePicker.onRtlPropertiesChanged (TimePicker.java:193)
  at android.view.View.resolvePadding (View.java:14637)
  at android.view.ViewGroup.resolvePadding (ViewGroup.java:6455)
  at android.view.View.initializeScrollbarsInternal (View.java:5019)
  at android.view.View.<init> (View.java:4438)
  at android.view.ViewGroup.<init> (ViewGroup.java:513)
  at android.widget.FrameLayout.<init> (FrameLayout.java:115)
  at android.widget.TimePicker.<init> (TimePicker.java:84)
  at android.widget.TimePicker.<init> (TimePicker.java:80)
  at android.widget.TimePicker.<init> (TimePicker.java:76)
```

![image](https://user-images.githubusercontent.com/41914678/74956248-ad048b00-53ba-11ea-9af1-9a806fb52e98.png)

**Comment by wouterattoptal** [comment]
And a slightly different looking one on another device:

```
android.view.InflateException: 
  at android.view.LayoutInflater.inflate (LayoutInflater.java:551)
  at android.view.LayoutInflater.inflate (LayoutInflater.java:429)
  at android.view.LayoutInflater.inflate (LayoutInflater.java:380)
  at android.app.TimePickerDialog.<init> (TimePickerDialog.java:152)
  at android.app.TimePickerDialog.<init> (TimePickerDialog.java:112)
  at crc643f46942d9dd1fff9.PickerManager_PickerListener.n_onClick (Native Method)
  at crc643f46942d9dd1fff9.PickerManager_PickerListener.onClick
  at android.view.View.performClick (View.java:5721)
  at android.widget.TextView.performClick (TextView.java:10936)
  at android.view.View$PerformClick.run (View.java:22620)
  at android.os.Handler.handleCallback (Handler.java:739)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:148)
  at android.app.ActivityThread.main (ActivityThread.java:7406)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1230)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1120)
Caused by: android.view.InflateException: 
  at android.view.LayoutInflater.createView (LayoutInflater.java:657)
  at com.android.internal.policy.PhoneLayoutInflater.onCreateView (PhoneLayoutInflater.java:58)
  at android.view.LayoutInflater.onCreateView (LayoutInflater.java:706)
  at android.view.LayoutInflater.createViewFromTag (LayoutInflater.java:774)
  at android.view.LayoutInflater.createViewFromTag (LayoutInflater.java:716)
  at android.view.LayoutInflater.inflate (LayoutInflater.java:498)
Caused by: java.lang.reflect.InvocationTargetException: 
  at java.lang.reflect.Constructor.newInstance (Native Method)
  at android.view.LayoutInflater.createView (LayoutInflater.java:631)
Caused by: java.lang.NullPointerException: 
  at android.widget.TimePicker.onRtlPropertiesChanged (TimePicker.java:248)
  at android.view.View.resolvePadding (View.java:15678)
  at android.view.ViewGroup.resolvePadding (ViewGroup.java:6974)
  at android.view.View.initializeScrollbarsInternal (View.java:5365)
  at android.view.View.<init> (View.java:4700)
  at android.view.ViewGroup.<init> (ViewGroup.java:591)
  at android.widget.FrameLayout.<init> (FrameLayout.java:97)
  at android.widget.TimePicker.<init> (TimePicker.java:78)
  at android.widget.TimePicker.<init> (TimePicker.java:74)
  at android.widget.TimePicker.<init> (TimePicker.java:70)

```

![image](https://user-images.githubusercontent.com/41914678/74956661-4895fb80-53bb-11ea-9ffe-4988a64d0795.png)

**Comment by ads90** [comment]
i have the same issue with Galaxy J7 and Galaxy A5 ios 6.0.1

**Comment by clintonrocksmith-skedulo** [comment]
We too have this problem

**Comment by clintonrocksmith-skedulo** [comment]
After a little investigation, I think it has something to do with the Context.  I made a custom Renderer to try and see if the values I was parsing were causing the issues.

I replaced the Context with a hack to get it working as seen in the code below.

This custom rendered inherits from TimePickerRenderer.

`new void CreateTimePickerDialog(int hours, int minutes) {
            _dialog = new TimePickerDialog(CrossCurrentActivity.Current.Activity, new TimePickerSetListener(this), new TimePickerClearer(this), 0, 0, false);
        }`

This seemed to work and I hope it helps bubble up the problem.

**Comment by FabienBehra38** [comment]
@clintonrocksmith-skedulo Thanks for sharing your code, but can you share the whole Renderer please ? I have errors on TimePickerSetListener and TimePickerClearer...

**Comment by kramer-e** [comment]
The same issue on a Galaxy A3 with Android 6.0.1. Fixed by using a custom renderer and using the current Activity, as clintonrocksmith-skedulo suggested.

```
protected override TimePickerDialog CreateTimePickerDialog(int hours, int minutes)
{
     return new TimePickerDialog(CrossCurrentActivity.Current.Activity, this, hours, minutes, true);
}
```

**Comment by Kikanye** [comment]
> The same issue on a Galaxy A3 with Android 6.0.1. Fixed by using a custom renderer and using the current Activity, as clintonrocksmith-skedulo suggested.
> 
> ```
> protected override TimePickerDialog CreateTimePickerDialog(int hours, int minutes)
> {
>      return new TimePickerDialog(CrossCurrentActivity.Current.Activity, this, hours, minutes, true);
> }
> ```

Please could you post the full code for your renderer? I just ran into this issue, and I am unable to find the fix.
Does this mean you have to use a custom TImePicker? (ie a custom control?)

**Comment by kramer-e** [comment]
@Kikanye Here is the full code, I did not add a custom TimePicker. I created a custom renderer for the TimePicker, so that each TimePicker rendered in Android uses this code with the override.

```
using System;
using Android.App;
using Android.Content;
using Plugin.CurrentActivity;
using <YourNamespaceHere>.Renderers;
using Xamarin.Forms;
using Xamarin.Forms.Platform.Android;
using static Android.App.TimePickerDialog;

[assembly: ExportRenderer(typeof(TimePicker), typeof(CustomTimePickerRenderer))]
namespace <YourNamespaceHere>.Renderers
{
    public class CustomTimePickerRenderer : TimePickerRenderer, IOnTimeSetListener
    {

        public CustomTimePickerRenderer(Context context) : base(context)
        {
        }

        protected override TimePickerDialog CreateTimePickerDialog(int hours, int minutes)
        {
            return new TimePickerDialog(CrossCurrentActivity.Current.Activity, this, hours, minutes, true);
        }
    }
}
```

**Comment by Kikanye** [comment]
> @Kikanye Here is the full code, I did not add a custom TimePicker. I created a custom renderer for the TimePicker, so that each TimePicker rendered in Android uses this code with the override.
> 
> ```
> using System;
> using Android.App;
> using Android.Content;
> using Plugin.CurrentActivity;
> using <YourNamespaceHere>.Renderers;
> using Xamarin.Forms;
> using Xamarin.Forms.Platform.Android;
> using static Android.App.TimePickerDialog;
> 
> [assembly: ExportRenderer(typeof(TimePicker), typeof(CustomTimePickerRenderer))]
> namespace <YourNamespaceHere>.Renderers
> {
>     public class CustomTimePickerRenderer : TimePickerRenderer, IOnTimeSetListener
>     {
> 
>         public CustomTimePickerRenderer(Context context) : base(context)
>         {
>         }
> 
>         protected override TimePickerDialog CreateTimePickerDialog(int hours, int minutes)
>         {
>             return new TimePickerDialog(CrossCurrentActivity.Current.Activity, this, hours, minutes, true);
>         }
>     }
> }
> ```

@kramer-e Thank you very much! this worked for me!

**Comment by EmilianoGallo** [comment]
someone have the solutions?? Android.Views.InflateException: 'Binary XML file line #20: Error inflating class android.widget.TimePicker'
WHEN I DELETE the scrollview Works fine, but when i put a  <ScrollView  > CRASH!!! help please

**Comment by samhouts** [comment]
@EmilianoGallo Can you please test with the latest 4.5.0 or 4.6.0 releases to see if the issue persists? Thanks!

**Comment by samhouts** [comment]
If you have an affected device, please also test the following package to see if this resolves the issue. Thank you! 

[Xamarin.Forms.4.7.0.2564.nupkg.zip](https://github.com/xamarin/Xamarin.Forms/files/4714022/Xamarin.Forms.4.7.0.2564.nupkg.zip)

**Comment by EmilianoGallo** [comment]
@samhouts i try to test 4.5.0 or 4.6.0 and crash  . Also i downgrade to 4.3.0.991250 and for me is not the solutions. My code is very simply <scrollview and TimePicker. When i click in TimePicker CRASH i have not code behind

**Comment by samhouts** [comment]
@EmilianoGallo What device? What API level is it?

**Comment by EmilianoGallo** [comment]
@samhouts  i test with the prerelase 4708013 pre2 i download from nuget. Also crash.
Xamarin.Forms.4.7.0.2564.nupkg.zip i dont know how to manual install sorry but if you tell me that i install and works i do it!.  What device? What API level is it? <uses-sdk android:minSdkVersion="21" android:targetSdkVersion="28" /> i test in 2 samssug A5 and galaxy prime

**Comment by samhouts** [comment]
@EmilianoGallo You can unzip the zip file to get the nupkg file, and put that somewhere on your computer. Then add a new package source, following [these instructions](https://docs.microsoft.com/en-us/nuget/consume-packages/install-use-packages-visual-studio#package-sources), and set the source as that folder.  Thanks!

**Comment by EmilianoGallo** [comment]
@samhouts YEEEEESSS with the Xamarin.Forms.4.7.0.2564.nupkg.zip WORK fine!!!!! iupiiii. Can i deploy this one in google play?? i have all my users down becaouse of this. Can i compile and make the pkg file and distribute? for all my users?? my app is in google play

**Comment by samhouts** [comment]
@EmilianoGallo That's great news! We don't typically recommend publishing apps with packages until they are available on NuGet. If you are able to wait about a week, we should have the fix fully tested and published live in a service release for 4.6.0. Thank you so much for testing the fix!

**Comment by EmilianoGallo** [comment]
@samhouts  I test my app and see two little visual bugs, but i dont care because the crash was fix it. I publish and send to the Play Store anyway . I will wait the oficial realease and then publish again. Thanks for help us!!! sorry for my bad english!

**Comment by samhouts** [comment]
@EmilianoGallo Can you send screenshots of the little visual bugs? Thanks!!

**Comment by EmilianoGallo** [comment]
<img width="270" alt="Screenshot_2020-06-02-14-35-47" src="https://user-images.githubusercontent.com/50379601/83551979-7244df80-a4df-11ea-8312-71c1c6d8ed8f.png">
<img width="270" alt="Screenshot_2020-06-02-14-36-49" src="https://user-images.githubusercontent.com/50379601/83551981-73760c80-a4df-11ea-896e-15d334d48546.png">

here are the screensshots, thanks

**Comment by EmilianoGallo** [comment]
Hi @samhouts This fixs is in progress yet? The scroll with timepicker bug and this little visual bugs??
Thanks! when will the release is ready to download? and where? in nuget? thanks!

**Comment by EmilAlipiev** [comment]
was that published in any version of 4.7.0? i cannot see in the release notes.
